### PR TITLE
mark deleted message as read, fix #634

### LINKED ISSUE #634: Unread counter is not decreased when an unread mail is deleted directly
1. Open a folder with a bunch of unread mails
2. Delete an unread mail without opening it
3. The unread counter does not decrease, should though

cc @DeepDiver1975 @wurstchristoph @zinks- @Xenopathic

### PR DESCRIPTION
**Pull request by jancborchardt**
Attempt to fix https://github.com/owncloud/mail/issues/634 but it doesn’t quite work.

What am I doing wrong @DeepDiver1975 @Xenopathic?

### CODE REVIEW

**Comment by clementhk** [comment]
I'm not following, why are they getting set to `unseen`?

**Comment by irgendwie** [comment]
@jancborchardt instead of your changes, add:
`thisModel.get('flags').set('unseen', false);`

**Comment by clementhk** [comment]
I don't think this is the right fix as we might know what it to be set as seen in trash?

**Comment by jancborchardt** [comment]
@clementhk yeah, was thinking that too. Will check with @DeepDiver1975

**Comment by jancborchardt** [comment]
Please review again @clementhk @Xenopathic @rullzer @zinks-

**Comment by jancborchardt** [comment]
Decreasing the counter on deleting unread mails works. But not when viewing them.

**Comment by jancborchardt** [comment]
@rullzer can you review this as well? It basically replaces https://github.com/owncloud/mail/pull/636

**Comment by jancborchardt** [comment]
@DeepDiver1975 the »stop unread counter from decreasing on star/unstar« fixed by https://github.com/owncloud/mail/pull/636 still occurs here.

**Comment by zinks-** [comment]
I confirm the (un)star decreases the counter

**Comment by DeepDiver1975** [comment]
@jancborchardt @zinks- please retest - THX

**Comment by jancborchardt** [comment]
@clementhk @irgendwie @zinks- @rullzer @wurstchristoph can you retest this? We should get it into 0.1 :)

File: js/views/message.js [hunk 1]
```diff
@@ -184,6 +187,9 @@ views.Messages = Backbone.Marionette.CompositeView.extend({
 	changeFlags: function(model) {
 		var unseen = model.get('flags').get('unseen');
 		var prevUnseen = model.get('flags')._previousAttributes.unseen;
+		//if(_.isUndefined(model._previousAttributes.flags.unseen)) {
+		//	prevUnseen = model._previousAttributes.flags.get('unseen');
+		//}
```
Comment (wurstchristoph) [inline js/views/message.js]: we should probably remove those commented lines ;-)

**Comment by wurstchristoph** [comment]
:+1:

File: js/views/message.js (hunk 1 above)
Comment (jancborchardt) [inline js/views/message.js]: @DeepDiver1975 do you still need them, or can you remove them? ;)

**Comment by irgendwie** [comment]
LGTM :+1:

**Comment by jancborchardt** [comment]
Merging for now, we can remove the commented stuff later.

**Pull request merged**